contact.php

```diff
@@ -68,85 +75,85 @@
 
 				$errors =  array('name' => '', 'email' => '', 'phone' => '', 'comment' => '', 'location' => '');
 
-				if (!empty($_POST["name"])) {
-					$name = trim($_POST['name']);
-				} else {
-					echo $errors['name'] = "<h2> Name is Required</h2>";
-				}
-				if (!empty($_POST["email"])) {
-					$email = trim($_POST['email']);
-				} else {
-					echo $errors['email'] = "<h2> Email is Required</h2>";
-				}
-				if (!empty($_POST["phone"])) {
-					$phone = trim($_POST['phone']);
-				} else {
-					echo $errors['phone'] = "<h2> Phone Number is Required</h2>";
-				}
-				if (!empty($_POST["comment"])) {
-					$comment = trim($_POST['comment']);
-				} else {
-					echo $errors['comment'] = "<h2> Please provide a comment</h2>";
-				}
-				if (!empty($_POST["location"])) {
-					$location = $_POST['location'][0];
-				} else {
-					echo $errors['location'] = "<h2> Location is Required</h2>";
-				}
-				if (!array_filter($errors)) {
-					echo "<h2>Thank you!</h2>";
-				}
-			}
-			?>
-		</div>
-		<div class="contactform">
-			<form action="contact.php" method="post">
-				<div class="cf-group">
-					<label>Name:</label><input type="text" name="name" value="<?= ($name ?? ''); ?>"></br>
-				</div>
-				<div class="cf-group">
-					<label>E-mail:</label><input type="text" name="email" value="<?= ($email ?? ''); ?>"></br>
-				</div>
-				<div class="cf-group">
-					<label>Phone:</label><input type="text" name="phone" value="<?= ($phone ?? ''); ?>"></br>
-				</div>
-				<div class="cf-group">
-					<label>Comment:</label><textarea type="text" name="comment" rows="5" cols="50"><?= ($comment ?? ''); ?></textarea></br>
-				</div>
-				<div class="cf-group">
-					<label>Location:</label><input type="radio" value="EA" name="location[]" <?= (!empty($location) && $location == "EA" ? 'checked' : ''); ?>>East Aurora
-					<input type="radio" value="CL" name="location[]" <?= (!empty($location) && $location == "CL" ? 'checked' : ''); ?>>Clarence </br>
-				</div>
-				<div class="cf-group">
-					<label for="myfile">Select a file:</label>
-					<input type="file" id="myfile" name="myfile"></br>
-				</div>
-				<input type="submit" class="button" name="submit" value="Submit">
-			</form>
-		</div>
-	</div>
-	<footer>
-		<div class="footer">
-			<h2>Our Locations</h2>
-			<p><em>East Aurora Takeout</em><br />
-				191 Main Street, Rear<br />
-				East Aurora, NY 14052<br />
-				(716) 652-7959
-			</p>
-			<p><em>East Aurora</em><br />
-				185 Main Street<br />
-				East Aurora, NY 14052<br />
-				(716) 652-7959
-			</p>
-			<p><em>Clarence</em><br />
-				8326 Main Street<br />
-				Clarence, NY 14221<br />
-				(716) 710-9464
-			</p>
-		</div>
-	</footer>
+                if (!empty(trim($_POST["name"]))) {
+                    $name = trim($_POST['name']);
+                } else {
+                    echo $errors['name'] = "<h2> Name is Required</h2>";
```

Comment:
I'm not sure you want to echo here... Though maybe you do... We should talk about your intentions with this line.

Reply:
This echo demonstrates to the user that a name is required at the top of the form after they hit submit and did not input a valid name , so it is easy to see. Hope that answers your question.